src/main/java/org/reaktivity/nukleus/maven/plugin/internal/ast/parse/AstParser.java

```diff
@@ -190,16 +195,12 @@ public AstEnumNode visitEnum_type(
     public AstValueNode visitEnum_value(
         Enum_valueContext ctx)
     {
-        AstValueNode.Builder valueBuilder = new AstValueNode.Builder();
-
         super.visitEnum_value(ctx);
-
-        AstValueNode value = valueBuilder.name(ctx.ID().getText())
-                                         .ordinal(enumBuilder.size())
-                                         .build();
-
+        AstValueNode.Builder valueBuilder = new AstValueNode.Builder();
+        valueBuilder.name(ctx.ID().getText());
+        AstValueNode value = enumBuilder.explicitType() ? valueBuilder.value(parseInt(ctx.uint_literal())).build()
+            : valueBuilder.value(enumBuilder.size()).build();
```

Comment:
Recommend you keep the `ordinal` in place on the `AstValueNode` but also introduce the `value` as an optional member, with `null` meaning unused during generation phase.
```
Integer parsed = null;
switch (enumBuilder.valueType())
{
case UINT32:
    parsed = ctx.uint_literal();
    break;
}

AstValueNode value = valueBuilder.name(ctx.ID().getText())
                                 .ordinal(enumBuilder.size())
                                 .value(parsed)
                                 .build();
```